src/mpid/ch4/src/ch4_progress.h

```diff
@@ -13,18 +13,24 @@
 
 #include "ch4_impl.h"
 
+#define MPID_CH4_PROGRESS_NBC  (1)
+#define MPID_CH4_PROGRESS_NM   (1<<1)
+#define MPID_CH4_PROGRESS_SHM  (1<<2)
+
+#define MPID_CH4_PROGRESS_ALL (MPID_CH4_PROGRESS_NBC|MPID_CH4_PROGRESS_NM|MPID_CH4_PROGRESS_SHM)
+
 #undef FUNCNAME
-#define FUNCNAME MPID_Progress_test
+#define FUNCNAME MPID_Progress_test_impl
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-MPL_STATIC_INLINE_PREFIX int MPID_Progress_test(void)
+MPL_STATIC_INLINE_PREFIX int MPID_Progress_test_impl(int flags)
 {
     int mpi_errno, made_progress, i;
 
-    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_PROGRESS_TEST);
-    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_PROGRESS_TEST);
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_PROGRESS_TEST_IMPL);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_PROGRESS_TEST_IMPL);
 
-    if (OPA_load_int(&MPIDI_CH4_Global.active_progress_hooks)) {
+    if (OPA_load_int(&MPIDI_CH4_Global.active_progress_hooks) && (flags & MPID_CH4_PROGRESS_NBC)) {
```

Comment:
This doesn't look right to me. We might have progress hooks unrelated to NBC.

Reply:
Oh, yes, that's true. Renamed that.

Reply:
@alexander-sannikov I think the rename was only half-committed, but after some more thought I think the proper namespace for this function would be `MPIDI_Progress_test`. This was it is clear it should only be used from within the device. Flags should be named `MPIDI_PROGRESS_HOOKS`, `MPIDI_PROGRESS_NM`, etc. Do you agree?

Reply:
@raffenet done.